x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_overview.tsx

```diff
@@ -36,22 +40,94 @@ export const CrawlerOverview: React.FC = () => {
       pageHeader={{ pageTitle: CRAWLER_TITLE }}
       isLoading={dataLoading}
     >
-      <EuiFlexGroup direction="row" alignItems="stretch">
-        <EuiFlexItem>
+      {domains.length > 0 ? (
+        <>
+          <EuiFlexGroup direction="row" alignItems="stretch">
+            <EuiFlexItem>
+              <EuiTitle size="s">
+                <h3>
+                  {i18n.translate('xpack.enterpriseSearch.appSearch.crawler.domainsTitle', {
+                    defaultMessage: 'Domains',
+                  })}
+                </h3>
+              </EuiTitle>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <AddDomainFlyout />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiSpacer size="m" />
+          <DomainsTable />
+        </>
+      ) : (
+        <>
           <EuiTitle size="s">
             <h3>
-              {i18n.translate('xpack.enterpriseSearch.appSearch.crawler.domainsTitle', {
-                defaultMessage: 'Domains',
+              {i18n.translate('xpack.enterpriseSearch.appSearch.crawler.empty.title', {
+                defaultMessage: 'Add a domain to get started',
```

Comment:
[Not a blocker, can wait until after 7.15] I would strongly consider using the AppSearchPageTemplate's `emptyState` and `isEmptyState` props to show this empty state (as well as using an `<EuiEmptyPrompt />` to match the rest of our app. Maybe @daveyholler can take a look at this and tweak it later once this PR lands? But ideally you'd pull this copy/documentation to its own empty state component and your code would look like this:

```jsx
    <AppSearchPageTemplate
      pageChrome={getEngineBreadcrumbs([CRAWLER_TITLE])}
      pageHeader={{ pageTitle: CRAWLER_TITLE }}
      isLoading={dataLoading}
      isEmptyState={!domains.length}
      emptyState={<EmptyState />}
    >
      // DomainsTable, no conditional needed
```

Hopefully cleaner all around to read and less conditionals

Reply:
Ahh gotcha, I totally missed that nuance, apologies.